Packs/ApiModules/Scripts/AggregatedCommandApiModule/AggregatedCommandApiModule.py

```diff
@@ -0,0 +1,989 @@
+            validate_input_function (Callable[[dict[str, Any]], None]): Function to validate the input, the function
+            should receive args, validate the inputs, and raise an error if the input is invalid. Defaults to lambda args: None.
+        """
+        super().__init__(args, brands, verbose, commands, validate_input_function)
+        self.external_enrichment = external_enrichment
+        self.final_context_path = final_context_path
+        self.additional_fields = additional_fields
+        self.data = data
+        self.indicator = indicator
+    
+    @cached_property
+    def external_missing_brands(self) -> list[str]:
+        """
+        Returns a list of external brands to run on from the given brands.
+        If no brands are given, returns empty list.
+        Caches the result to avoid redundant calculations.
+        """
+        if not self.brands:
+            demisto.debug("No specific brands provided; will run on all available brands.")
+            return []
+        requested = set(self.brands)
+        non_external_brands = {command.brand for command in self.commands if command.command_type != CommandType.EXTERNAL}
+        external_brands = requested - non_external_brands
+        demisto.debug(f"External brands to run on: {external_brands}")
+        return list(external_brands - set(self.enabled_brands))
+    
+    def aggregated_command_main_loop(self) -> CommandResults:
+        """
+        Main execution loop for the reputation aggregation.
+        """
+        demisto.debug("Starting aggregated command main loop.")
+        
+        # 1. Prepare commands and get TIM results
+        demisto.debug("Step 1: querying TIM.")
+        tim_context, tim_dbot, tim_entries = self.get_indicators_from_tim()
+        demisto.debug(f"TIM query resulted in {len(tim_dbot)} DBot scores, "
+                      f"and {sum(len(lst) for brands_map in tim_context.values() for lst in brands_map.values())} indicators.")
+        
+        # 2. Execute batch commands
+        demisto.debug("Step 2: Executing batch commands.")
+        commands_to_execute = self.prepare_commands(self.external_enrichment)
+        if commands_to_execute:
+            batch_executor = BatchExecutor(commands_to_execute, self.brands_to_run)
+            batch_results = batch_executor.execute()
+            demisto.debug(f"Batch execution resulted in {len(batch_results)} results.")
+        else:
+            demisto.debug("No commands to execute.")
+            batch_results = []
+        
+        # 3. Process batch results
+        demisto.debug("Step 3: Processing batch results.")
+        batch_context, batch_dbot, verbose_outputs, batch_entries = self.process_batch_results(batch_results, commands_to_execute)
+        
+        # 4. Merge context
+        demisto.debug("Step 4: Merging all contexts.")
+        final_context = self.merge_context(tim_context, batch_context,tim_dbot, batch_dbot)
```

Comment:
reccomend having output context be a builder - its own class - to handle its own logic about how to merge/deal with context.
Usage 
```
ContextBuilder builder = new ContextBuilder()

self.builder.addContext(...)
self.builder.addContext(...)
final_context = self.builder.getFinalContext()
```
fyi @joeymizrahi